packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -720,6 +720,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             storage,
         );
 
+        // logging hardware telemetry
+        logger.sendTelemetryEvent({
+            eventName:"hardware",
+            // eslint-disable-next-line @typescript-eslint/dot-notation
+            deviceMemory: Object.hasOwnProperty.call(navigator, "deviceMemory") ? navigator["deviceMemory"] : undefined,
```

Comment:
Is this the same as just doing
deviceMemory: navigator?.deviceMemory

Reply:
That said, if you want to do check like you do, I'd rather do this check
("deviceMemory" in navigator)

The difference between this and just logging navigator?.deviceMemory does not exist, as the former will return true if property exists, including if it exists and undefined, but result of logging undefined is the same - we do not log it

Reply:
it's not the same because using `.deviceMemory` will produce error "deviceMemory does not exist on type Navigator"

Reply:
Can you please clarify? These things work fine in Node:
{}?.navigator
undefined?.deviceMemory
null?.deviceMemory

What does not work in Node is navigator?.deviceMemory which should not happen with you check of typeof (though I'd recommend to do positive check, i.e. (typeof navigator === "object") - it will only allow objects & null through, which is stricter.

Reply:
it's a compile error. when I do navigator?.deviceMemory, it says "deviceMemory does not exist on type Navigator"

Reply:
Got it. Then I guess (navigator as any)?.deviceMemory would be a bit more concise